src/test/java/org/fcrepo/client/ConnectionManagementIT.java

```diff
@@ -0,0 +1,371 @@
+import java.lang.reflect.Method;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import static org.fcrepo.client.MockHttpExpectations.host;
+import static org.fcrepo.client.MockHttpExpectations.port;
+import static org.fcrepo.client.TestUtils.TEXT_TURTLE;
+import static org.fcrepo.client.TestUtils.setField;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+/**
+ * Integration test used to demonstrate connection management issues with the FcrepoClient.
+ *
+ * @author esm
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class ConnectionManagementIT {
+
+    /**
+     * Starts a mock HTTP server on a free port
+     */
+    @Rule
+    public MockServerRule mockServerRule = new MockServerRule(this);
+
+    // Set by the above @Rule, initialized on @Before via MockHttpExpectations
+    private MockServerClient mockServerClient;
+
+    /**
+     * URIs that our Mock HTTP server responds to.
+     */
+    private MockHttpExpectations.SupportedUris uris;
+
+    /**
+     * Verifications for HttpClientConnectionManager
+     */
+    static class Verifications {
+
+        /**
+         * Verifies that the expected number of connections have been requested and then closed.
+         *
+         * @param connectionCount the number of connections that have been requested and closed.
+         * @return a Consumer that verifies the supplied HttpClientConnectionManager has opened and closed the expected
+         * number of connections.
+         */
+        static Consumer<HttpClientConnectionManager> connectionRequestedAndClosed(final int connectionCount) {
+            return (connectionManager) -> {
+                // A new connection was requested by the Http client ...
+                verify(connectionManager, times(connectionCount)).requestConnection(any(HttpRoute.class), any());
+
+                // Verify it was released.
+                verify(connectionManager, times(connectionCount)).
+                        releaseConnection(any(HttpClientConnection.class), any(), anyLong(), any(TimeUnit.class));
+            };
+        }
+
+        /**
+         * Verifies that the expected number of connections have been requested and <em>have not been</em> closed.
+         *
+         * @param connectionCount the number of connections that have been requested.
+         * @return a Consumer that verifies the supplied HttpClientConnectionManager has opened the expected
+         * number of connections, but hasn't closed <em>any</em> of them
+         */
+
+        static Consumer<HttpClientConnectionManager> connectionRequestedButNotClosed(final int connectionCount) {
```

Comment:
It's not totally clear to me where the value of making this a `Consumer` (and not just a `static void` method with a `HttpClientConnectionManager` param) is coming in?